src/krb5/_creds.pyx

```diff
@@ -97,6 +167,68 @@ cdef class Creds:
     def __str__(Creds self) -> str:
         return "Creds"
 
+    @property
+    def client(Creds self) -> Principal:
+        princ = Principal(self.ctx, 0, needs_free=0)
+        pykrb5_creds_get(&self.raw, &princ.raw, NULL, NULL, NULL, NULL, NULL, NULL)
+
+        return princ
+
+    @property
+    def server(Creds self) -> Principal:
+        princ = Principal(self.ctx, 0, needs_free=0)
+        pykrb5_creds_get(&self.raw, NULL, &princ.raw, NULL, NULL, NULL, NULL, NULL)
+
+        return princ
+
+    @property
+    def keyblock(Creds self) -> KeyBlock:
+        kb = KeyBlock(self.ctx, needs_free=0)
```

Comment:
Need to figure out if it's possible to copy this data to avoid the deallocation problem. I've already made the mistake of not doing so for other attributes that return a KeyBlock.

Reply:
This should be possible simply by calling [`krb5_copy_keyblock()`](https://web.mit.edu/kerberos/krb5-latest/doc/appdev/refs/api/krb5_copy_keyblock.html) I think. I can try to do that.